### DISCUSSION TITLE
How to setup a Gitpod workspace with Tasmota Previous Release Source tar archive

### DISCUSSION THREAD

**Discussion by jkwim** [arendst/Tasmota#20698] [Q&A] [answered]
I would like to know how to setup a new Workspace in Gitpod with Tasmota Source of a previous release?

Say I have downloaded the source files of Tasmota-13.1.0 and would like to have a workspace setup for custom compilation of this version.

The New Workspace creation option in Gitpod requires a URL for the repository.

Instead if I want to untar the source archive in to the workspace how should I proceed?

**Comment by barbudor** [comment]
Gitpod is meant to be used from a online repo, not from files on your computer.

If you want to use the tar that you downloaded, get VSCode on your computer

Otherwise just proceed from Github to launch Gitpod and from within Gitpod select the branch/tag/commit from within Gitpod UI

**Comment by jkwim** [comment, marked as answer]
Do you mean like this?

![image](https://github.com/arendst/Tasmota/assets/84637540/2d609da1-4416-4e9f-b64e-6872fcb78f4f)

**Reply by barbudor** [reply]
Yes

**Reply by jkwim** [reply]
What are the steps to make it my own workspace only.

I am getting confused with this status

![image](https://github.com/arendst/Tasmota/assets/84637540/388cffe1-8389-4c4a-a07a-08a271264413)

**Reply by barbudor** [reply]
there are 2 diffirent thing that you may mistaking
- the gitpod workspace
- the repository

if you want to modify Tasmota and keep the changes, you should 1st fork Tasmota into your own Github account then start Gitpod from there (jkwim/Tasmota) , and not from arenstd/Tasmota
So if you make changes, you can push (save) those changes in your fork. Preferably use a dedicated branch if you later want to merge in changes from Tasmota

keeping the gitpod workspace valid, you can use the 3 dots menu to make it persistent so it doesn't disapear.

**Comment by jkwim** [comment]
Apologize for being naive on this subject. I do not come from a professional software developer background.

What I want is to maintain my custom compilation environment and maintain that separately for say two Tasmota Releases + Development branch.

I understand the forking at a particular ** release commit ** but yet to figure out the steps to do it on Github.

So in order to have two (three) compilation environments do I have to have 3 x forks then?

Only the development fork would need to get updates from the main repository. Other two will be static, code wise, once forked.

Something like this:

```
Development
         ^
	 |
         O----->O Sync into private fork
	 |      |
	 |      |
	 |      |
         O----->O Sync into private fork
	 |      |
	 |      |
	 |      |
	 |      |
         O----->O Sync into private fork
	 |      |
	 |      |
	 |      |
	 |      |
	 |      |
	 |      |
         O------O  Fork from Development 
	 |
	 O------O  Fork2 (Rel 13.1.0)
	 |
	 |
	 |
	 |
	 |
	 O------O  Fork1 (Rel 12.3.1)
	 |
	 |
	 |
arenstd/Tasmota	 
```
